### PR TITLE
Feat: new `RenderSections` for all pages

### PR DESCRIPTION
**Pull request by eduardoformiga**
Depends on 
- https://github.com/vtex/faststore/pull/2555

## What's the purpose of this pull request?

This PR is part of the performance initiative. 

It aims to create a new structure for the `RenderSections` that uses the `ViewportObserver` along with the `LazyLoadingSection` to load the section on demand. Now, the RenderSections receives the `globalSection` as props, and we are removing the `GlobalSection` (parent) Component that wraps the pages.

It also applies this structure to all pages:
- home
- pdp
- plp
- login
- checkout
- account
- search
- landing page
- 500
- 404

PS:  some changes from PLP and Landing Page need to be applied now. Then I applied some `Promise.all` and other improvements beforehand.

## How to test it?

Now, we can only load the JS for the sections inside the viewport during the first load. You can check the network tab in devtools while scrolling the page until the `Footer` to double-check the JS for sections above the fold being loaded.

You can run the project locally or use the starter preview to double-check this behavior.

you can test all the pages. The landing page can be accessed through this URL:
http://localhost:3000/my-landing-page

https://github.com/user-attachments/assets/8d69faa5-782f-4fb4-81fc-fd637043870b

## Performance

||Before|After|
|-|-|-|
||https://starter.vtex.app/|https://sfj-b662675--starter.preview.vtex.app/|
|on a lucky day (when PSI has a good sense of humor, haha)|Screenshot 2024-11-12 at 19 49 50|Screenshot 2024-11-12 at 19 50 00|
|on a normal day|Screenshot 2024-11-12 at 19 54 24|Screenshot 2024-11-12 at 19 54 17|

### Starters Deploy Preview

- https://github.com/vtex-sites/starter.store/pull/612

preview 
https://sfj-b662675--starter.preview.vtex.app/

## References

In this video, you can see an explanation of loading JS on demand using the Intersection Observer API + code splitting.
 https://drive.google.com/file/u/1/d/13VYPu45Vaav2ecGmKcFAOUMKAS3pMHd4/view

POC PR
- https://github.com/vtex/faststore/pull/2404

### CODE REVIEW

File: packages/core/src/pages/s.tsx [hunk 1]
```diff
@@ -70,7 +70,7 @@ function Page({ page: searchContentType, globalSections }: Props) {
   }
 
   return (
-    <GlobalSections {...globalSections}>
+    <>
```
[the remark this answers is not in the document]
Comment (eduardoformiga) [reply packages/core/src/pages/s.tsx]: True! I removed the fragment [here](https://github.com/vtex/faststore/pull/2558/commits/2cd3e9743b729f22868500b32fa02904bad38759) ty :)

File: packages/core/src/components/cms/plp/Components.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import { OverriddenDefaultProductGallery as ProductGallery } from 'src/components/sections/ProductGallery/OverriddenDefaultProductGallery'
+import CUSTOM_COMPONENTS from 'src/customizations/src/components'
+import { default as GLOBAL_COMPONENTS } from '../global/Components'
+
+const BannerText = dynamic(
+  () =>
+    import(
+      /* webpackChunkName: "BannerText" */
+      'src/components/sections/BannerText/OverriddenDefaultBannerText'
+    ).then((mod) => ({ default: mod.OverriddenDefaultBannerText })),
```
[the remark this answers is not in the document]
Comment (eduardoformiga) [reply packages/core/src/components/cms/plp/Components.ts]: yes! I also change it in the other files [here](https://github.com/vtex/faststore/pull/2558/commits/986e298e7bb9346d1aaeb7e3382abe82fd4b121b).

File: packages/core/src/components/templates/SearchPage/SearchPage.tsx [hunk 3]
```diff
@@ -1,46 +1,25 @@
...
   useCreateUseGalleryPage,
   UseGalleryPageContext,
 } from 'src/sdk/product/usePageProductsQuery'
 import { SearchContentType } from 'src/server/cms'
-import type { ClientProductGalleryQueryQuery as ClientProductGalleryQuery } from '@generated/graphql'
-
-/**
- * Sections: Components imported from each store's custom components and '../components/sections' only.
- * Do not import or render components from any other folder in here.
- */
```
[the remark this answers is not in the document]
Comment (eduardoformiga) [reply packages/core/src/components/templates/SearchPage/SearchPage.tsx:25]: Yes! I agree. Done [here](https://github.com/vtex/faststore/pull/2558/commits/e2ccf30c1a6bece42761ed2c167714e9b86172bc).

File: packages/core/src/pages/[...slug].tsx [hunk 4]
```diff
@@ -25,6 +26,10 @@ import { PageContentType } from 'src/server/cms'
 import { getPLP, PLPContentType } from 'src/server/cms/plp'
 import { getDynamicContent } from 'src/utils/dynamicContent'
 
+const LandingPage = dynamic(
```
[the remark this answers is not in the document]
Comment (eduardoformiga) [reply packages/core/src/pages/[...slug].tsx:29]: Good question!
While testing the performance from PLP, I noticed that both templates were being imported statically. As the PLP is more common than the Landing Page (LP), we need to make a decision and just split the code and take the JS from LP out, which brings a better score for PLP (more common page).

**Review by lariciamota — APPROVED** [review]

**Review by lucasfp13 — APPROVED** [review]

**Pull request merged**